internal/handler/api.go

```diff
@@ -19,11 +29,15 @@ func NewAPI(cfg config.Config, authService AuthService, userService UserService,
 				auth.POST("/signin", SigninHandler(authService, userService, cfg))
 				auth.POST("/reset-password", ResetPasswordHandler(userService))
 			}
-			users := v1.Group("/users").Use(middleware.JWTAuth(cfg.JWTTokenSecret))
+			users := v1.Group("/users")
 			{
+				admin := users.Use(middleware.JWTAuth(cfg.JWTTokenSecret), middleware.AccessCheck(adminRole))
+				{
+					admin.GET("", GetAllUsersHandler(userService))
```

Comment:
I would suggest naming "GetUsers" instead, since we will probably have some filters applied there (at the bare minimum pagination is applied, so by definition its not "All" users)